packages/wrangler/src/__tests__/publish.test.ts

```diff
@@ -7117,3 +7233,48 @@ function mockServiceScriptData(options: {
 		);
 	}
 }
+
+function mockGetQueue(expectedQueueName: string) {
+	const requests = { count: 0 };
+	setMockResponse(
+		`/accounts/:accountId/workers/queues/${expectedQueueName}`,
+		"GET",
+		([_url, accountId]) => {
+			expect(accountId).toEqual("some-account-id");
+			requests.count += 1;
+			return { queue: expectedQueueName };
+		}
+	);
+	return requests;
+}
+
+function mockGetQueueMissing(expectedQueueName: string) {
+	const requests = { count: 0 };
+	setMockResponse(
+		`/accounts/:accountId/workers/queues/${expectedQueueName}`,
+		"GET",
+		([_url, _accountId]) => {
+			throw { code: 100123 };
+		}
+	);
+	return requests;
+}
+
+function mockPutQueueConsumer(
+	expectedQueueName: string,
+	expectedConsumerName: string,
+	expectedBody: PutConsumerBody
+) {
+	const requests = { count: 0 };
+	setMockResponse(
+		`/accounts/:accountId/workers/queues/${expectedQueueName}/consumers/${expectedConsumerName}`,
+		"PUT",
+		([_url, accountId], { body }) => {
+			expect(accountId).toEqual("some-account-id");
+			expect(JSON.parse(body as string)).toEqual(expectedBody);
+			requests.count += 1;
+			return { queue: expectedQueueName };
+		}
+	);
+	return requests;
+}
```

Comment:
There is some tooling called MSW (Mock Service Worker) we are moving over too in the codebase, eventually removing the custom mocking around fetch & requests, if you could utilize that it would be great. (non-blocking for end of week)

Reply:
100% More than happy to pair on that 😃